### ISSUE TITLE
Default "order by"

### ISSUE THREAD

**Issue by peterdesmet** [inbo/etn#121]
Currently, data are returned as they are in the view. This is not always the most convenient for users to explore. Should they be returned with a default sort? @IPauwels @jreubens @PieterjanVerhelst 

Conclusion:

- [x] animals: by `animal_project_code`, `release_date_time` and then `tag_id`. Same is used by Movebank
- [x] tags: by `tag_id`
- [x] detections: by `tag_id` and then `date_time`. Same is used by Movebank
- [x] deployments: `network_project_code`, `station_name`, `deploy_date_time`
- [x] receivers: by `receiver_id`
- [x] projects: by `project_code`

**Comment by IPauwels** [comment]
This would be my suggestion:
- animals by release_date_time
- tags by tag_id 
- detections by date_time
- deployments by station_name?
- receivers by receiver_id

**Comment by jreubens** [comment]
I agree with Ine

**Comment by peterdesmet** [comment]
There is also `get_projects()`. Should this be by `project_code` or `start_date` (not always populated)

**Comment by IPauwels** [comment]
- Detections: that is fine by me and a good idea
- Deployments: I think I would like it as a user to have the deployments sorted as how they are geographically located, but I understand that consistency is also important, so then I would prefer perhaps indeed receiver_id. It is up to the scientist then to account for the fact that a minority of them might have changed location during the project.

Regarding get_projects(), I would say project_code. 

I think for a lot of cases Movebank is a good reference.

**Comment by jreubens** [comment]
For get_projects, I agree with project_code.
For detections I agree as well
For Deployments however, it occurs that several receivers are used at one station (in different deployments).
Sorting by station_name and then on date_time seems more logic to me. But I guess it won't differ too much...

**Comment by peterdesmet** [comment]
That's two votes for sorting by station_name, which makes sense, as it is geographical.

1. One issue is a bit that `station_names` are not very controlled, so sorting by `station_name` and `deploy_date_time` mixes projects:

pk | receiver_id | network_project_code | station_name | station_description | deploy_date_time
--- | --- | --- | --- | --- | ---
5552 | VR2-6481 | EMMN | 12 | 12 | 2007-04-25T00:00:00Z
4274 | VR2W-102239 | Artevigo | 12 | vinhos 12 | 2011-03-01T00:00:00Z
4194 | VR2W-119113 | OTN-Tosenfjorden | 12 | Tosenfjorden | 2015-04-07T08:00:00Z
4116 | VR2W-108576 | OTN-Skjerstadfjorden | 12 | v2lchases | 2016-05-02T08:00:00Z
4362 | HR2-461111 | PhysFish | 12 | Dubh Loch | 2016-06-01T00:00:00Z
4117 | VR2W-AR-546359 | OTN-Skjerstadfjorden | 12 | v2lchases | 2017-05-03T08:00:00Z
4308 | VR2W-131991 | paintedcomber | 12 | MPA Palma | 2017-07-12T12:00:00Z
4379 | VR2AR-546175 | Aberdeen | 12 | Aberdeen Bay | 2017-07-14T13:28:00Z
5947 | TBR 700-SN000152 | NTNU-Gaulosen | 12 | Gaulosen 12 | 2018-01-31T18:00:00Z
5053 | VR2W-133611 | Inforbiomares | 12 | 12 | 2018-07-23T11:30:00Z
6085 | VR2Tx-482283 | SPAWNSEIS | 12 | Bakkasund | 2018-10-24T07:56:20Z
8975 | VR2W-113599 | CONNECT-MED | 12 | 12 | 2019-05-23T00:00:00Z
5066 | VR2W-133600 | Inforbiomares | 12 | 12 | 2019-05-25T00:00:00Z
6501 | VR2Tx-482283 | SPAWNSEIS | 12 | Bakkasund | 2019-06-04T00:00:00Z
6317 | VR2W-133600 | Inforbiomares | 12 | 12 | 2019-11-07T00:00:00Z

2. We could sort by `station_name`, `network_project_code` and then `deploy_date_time`:

pk | receiver_id | network_project_code | station_name | station_description | deploy_date_time
--- | --- | --- | --- | --- | ---
4379 | VR2AR-546175 | Aberdeen | 12 | Aberdeen Bay | 2017-07-14T13:28:00Z
4274 | VR2W-102239 | Artevigo | 12 | vinhos 12 | 2011-03-01T00:00:00Z
8975 | VR2W-113599 | CONNECT-MED | 12 | 12 | 2019-05-23T00:00:00Z
5552 | VR2-6481 | EMMN | 12 | 12 | 2007-04-25T00:00:00Z
5053 | VR2W-133611 | Inforbiomares | 12 | 12 | 2018-07-23T11:30:00Z
5066 | VR2W-133600 | Inforbiomares | 12 | 12 | 2019-05-25T00:00:00Z
6317 | VR2W-133600 | Inforbiomares | 12 | 12 | 2019-11-07T00:00:00Z
5947 | TBR 700-SN000152 | NTNU-Gaulosen | 12 | Gaulosen 12 | 2018-01-31T18:00:00Z
4116 | VR2W-108576 | OTN-Skjerstadfjorden | 12 | v2lchases | 2016-05-02T08:00:00Z
4117 | VR2W-AR-546359 | OTN-Skjerstadfjorden | 12 | v2lchases | 2017-05-03T08:00:00Z
4194 | VR2W-119113 | OTN-Tosenfjorden | 12 | Tosenfjorden | 2015-04-07T08:00:00Z
4308 | VR2W-131991 | paintedcomber | 12 | MPA Palma | 2017-07-12T12:00:00Z
4362 | HR2-461111 | PhysFish | 12 | Dubh Loch | 2016-06-01T00:00:00Z
6085 | VR2Tx-482283 | SPAWNSEIS | 12 | Bakkasund | 2018-10-24T07:56:20Z
6501 | VR2Tx-482283 | SPAWNSEIS | 12 | Bakkasund | 2019-06-04T00:00:00Z

3. Or sort on `network_project_code` first, and then `station_name` and `deploy_date_time`.

4. Or sort by `receiver_id`.

What has your preference?

**Comment by damianooldoni** [comment]
For deployments, I vote for option 2: sort by `station_name`, `network_project_code `and then `deploy_date_time`.
For the others (animals, ...) I agree 100%

**Comment by peterdesmet** [comment]
I think what will us closest to the expectations is:

- animals: by `animal_project_code`, `release_date_time`, `tag_id`
- deployments: by `network_project_code`, `station_name`, `deploy_date_time`

Very often you will filter by project first and then only see the second filtering.

**Comment by damianooldoni** [comment]
This is even better, especially if we think that network/animal project code is an argument of the functions to get deployments/animals. So, contrarily to what I wrote above, sorting by project code first seems the most natural option. 

I also gave a look to station names, naming them is totally uncontrolled as @peterdesmet already mentioned: I found even the `""` (empty string) value as station name! I wonder then how meaningful is ordering by station names? Possible alternative: `receiver_id` instead of `station_name` as second ordering field?

**Comment by peterdesmet** [comment]
Thanks for the feedback! @damianooldoni regarding `station_name`: see discussion above, even though it's not super controlled, it is preferred to group by geographical unit (station) rather than receiver_id. Within a project there are seldom duplicates of station names

**Comment by peterdesmet** [comment]
@IPauwels @jreubens @PieterjanVerhelst One remaining issue @damianooldoni and I have been trying to sort out (pun not intended) is the default sorting of `tag_id` (and to lesser extend `station_name` and `receiver_id`), which is:

```yaml
# default sorting
A69-1008-1
A69-1008-10
A69-1008-100
A69-1008-12
A69-1008-2
```

The more natural order you would expect is:

```yaml
# natural sorting
A69-1008-1
A69-1008-2
A69-1008-10
A69-1008-12
A69-1008-100
```

But we haven't found an elegant solution yet that does not affect the speed of the functions. Is it worth pursuing further to get the second sorting order? Note: `list_receiver_id()` function (and similar) now uses the natural sorting.

**Comment by jreubens** [comment]
I also prefer the natural sorting...

**Comment by PieterjanVerhelst** [comment]
Here the same; I prefer the natural sorting.

**Comment by peterdesmet** [comment]
Ok, it took me a while to figure it out, but we now have an elegant solution using `str_sort()` and `arrange()` in d2134eb. The order of `tag_ids`, `station_names` or `receiver_ids` will now be the same when using `list_` or `get_` functions.

**Comment by peterdesmet** [comment]
This is now implemented in #123 and version `0.1.0.9123` of the package. Thanks for all the feedback.
